src/sdl2_compat.c

```diff
@@ -5593,15 +5593,94 @@ static void SynchronizeEnvironmentVariables()
     SDL3_DestroyEnvironment(fresh_env);
 }
 
+static SDL_InitFlags PreInitSubsystem(SDL_InitFlags flags)
+{
+    /* If the subsystem is already initialized, mask out the flag for it */
+    flags &= ~SDL_WasInit(flags);
```

Comment:
This isn't quite correct. This will prevent SDL_InitSubSystem(SDL_INIT_VIDEO) twice from incrementing the refcount twice and the first SDL_QuitSubSystem(SDL_INIT_VIDEO) will shut down the video subsystem.

Reply:
This masked flag value is only used for determining which subsystem pre/post initialization steps we need to perfom (only when the subsystem goes from refcount 0->1). We're still passing the caller's unmodified flags value to `SDL3_InitSubsystem()`, so the subsystem refcount should still be accurate.

Reply:
Ah, okay.